src/main/java/org/opentripplanner/apis/gtfs/datafetchers/LegImpl.java

```diff
@@ -324,4 +326,13 @@ public DataFetcher<Iterable<Leg>> nextLegs() {
   public DataFetcher<Double> accessibilityScore() {
     return environment -> NumberMapper.toDouble(getSource(environment).accessibilityScore());
   }
+
+  @Override
+  public DataFetcher<Relay.ResolvedGlobalId> id() {
+    return environment -> {
+      var ref = getSource(environment).getLegReference();
+      var id = (ref == null) ? "" : LegReferenceSerializer.encode(ref);
```

Comment:
Yesterday in the developer meeting we thought it might make sense to use some special string like "NotAvailable" instead of "null" or empty string.